lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -148,21 +148,24 @@ def delete(self, uuid):
 class ApplicationResource(Resource):
     resource_class = tacp.ApplicationsApi
 
-    def filter(self, **filters):
-        return self.api.get_applications_using_get(
-            **self.get_filters_kws(**filters)
-        )
+    def filter(self, as_dict=False, **filters):
+        results = self.api.get_applications_using_get(
+            **self.get_filters_kws(**filters))
+        if as_dict:
+            return [result.to_dict() for result in
+                    results]
+        return results
 
     def get_by_uuid(self, uuid):
-        return self.api.get_application_using_get(uuid)
+        return self.api.get_application_using_get(uuid).to_dict()
```

Comment:
What does `self.api.get_application_using_get(uuid)` return if the uuid is not found? This might throw an exception if None is returned.